### PR TITLE
Refactoring Behaviour Tree

### PR DESCRIPTION
**Pull request by ipdae**
- Add BattleLog events in BT.Tick once

### CODE REVIEW

File: Lib9c/Battle/StageSimulator.cs [hunk 1]
```diff
@@ -146,6 +147,10 @@ public override Player Simulate()
                     var turnBefore = TurnNumber;
 #endif
                     character.Tick();
+                    while (!character.TurnEnd)
+                    {
+                        Thread.Sleep(100);
```
[the remark this answers is not in the document]
Comment (ipdae) [reply Lib9c/Battle/StageSimulator.cs:152]: Shall we fix it to wait inside `BT.Tick()`?

[the remark this answers is not in the document]
Comment (longfin) [reply Lib9c/Battle/StageSimulator.cs:152]: No, what I questioned was not where to do this `Sleep()`, but whether the waiting itself was needed.

As far as I know, BT is a purely deterministic algorithm, but I think this determinism can be broken if logic such as waiting for a certain amount of time enters.

File: .Lib9c.Tests/Model/PlayerTest.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        public void TickAlive()
+        {
+            var simulator = new StageSimulator(_random, _avatarState, new List<Guid>(), 1, 1, _tableSheets);
+            var player = simulator.Player;
+            var enemy = new Enemy(player, _tableSheets.CharacterSheet.Values.First(), 1);
+            player.Targets.Add(enemy);
+            player.InitAI();
+            player.Tick();
+
+            Assert.True(simulator.Log.Any());
```
Comment (longfin) [inline .Lib9c.Tests/Model/PlayerTest.cs:38]:
```suggestion
            Assert.NotEmpty(simulator.Log);
```

File: .Lib9c.Tests/Model/PlayerTest.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            var enemy = new Enemy(player, _tableSheets.CharacterSheet.Values.First(), 1);
+            player.Targets.Add(enemy);
+            player.InitAI();
+            player.CurrentHP = -1;
+
+            Assert.True(player.IsDead);
+
+            player.Tick();
+
+            Assert.True(simulator.Log.Any());
```
Comment (longfin) [inline .Lib9c.Tests/Model/PlayerTest.cs:56]:
```suggestion
            Assert.NotEmpty(simulator.Log);
```

**Review by longfin — APPROVED** [review]

**Review by longfin — APPROVED** [review]
Could you rebase or merge main to this branch?

**Pull request merged**